ui/src/ensembler/components/table/ListRouterVersionsForEnsemblerTable.js

```diff
@@ -51,6 +51,9 @@ export const ListRouterVersionsForEnsemblerTable = ({
   useEffect(() => {
     if (results.activeItems.length > 0){
       setEnsemblerUsedByActiveRouterVersion(true)
+      // If there is an active router version using this ensembler, it is definitely also a current router version using
+      // this ensembler
+      setEnsemblerUsedByCurrentRouterVersion(true)
```

Comment:
This change keeps the `ensemblerUsedByCurrentRouterVersion` state consistent with the actual status of the ensembler (as of present, there's a `modalClosed` function that unsets this state to `false` whenever the modal is closed - previously it was not guaranteed that this state would even get updated, especially if the condition `if (results.activeItems.length > 0)` was fulfilled), which allows the effect here https://github.com/caraml-dev/turing/pull/352/files#diff-2852d3e270c0d9ad9c03a7cfffb3edc5b9723d381090490fa33bd7496697eea7R26 to update the `canDeleteEnsembler` state correctly.